src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -0,0 +1,774 @@
+            this._rowMatchConditions = Enumerable.Range(0, BatchSize)
+                .Select(index => string.Join(" AND ", primaryKeyColumns.Select(col => $"{col} = @{col}_{index}")))
+                .ToList();
+
+            this._executor = executor;
+            this._logger = logger;
+
+            this._cancellationTokenSourceCheckForChanges = new CancellationTokenSource();
+            this._cancellationTokenSourceRenewLeases = new CancellationTokenSource();
+            this._cancellationTokenSourceExecutor = new CancellationTokenSource();
+
+            this._rowsLock = new SemaphoreSlim(1);
+            this._rows = new List<IReadOnlyDictionary<string, string>>();
+            this._leaseRenewalCount = 0;
+            this._state = State.CheckingForChanges;
+
+#pragma warning disable CS4014 // Queue the below tasks and exit. Do not wait for their completion.
+            _ = Task.Run(() =>
+            {
+                this.RunChangeConsumptionLoopAsync();
+                this.RunLeaseRenewalLoopAsync();
+            });
+#pragma warning restore CS4014
+        }
+
+        /// <summary>
+        /// Stops the change monitor which stops polling for changes on the user's table. If the change monitor is
+        /// currently executing a set of changes, it is only stopped once execution is finished and the user's function
+        /// is triggered (whether or not the trigger is successful).
+        /// </summary>
+        public void Stop()
+        {
+            this._cancellationTokenSourceCheckForChanges.Cancel();
+        }
+
+        public void Dispose()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Executed once every <see cref="PollingIntervalInSeconds"/> period. If the state of the change monitor is
+        /// <see cref="State.CheckingForChanges"/>, then the method query the change/worker tables for changes on the
+        /// user's table. If any are found, the state of the change monitor is transitioned to
+        /// <see cref="State.ProcessingChanges"/> and the user's function is executed with the found changes. If the
+        /// execution is successful, the leases on "_rows" are released and the state transitions to
+        /// <see cref="State.CheckingForChanges"/> once again.
+        /// </summary>
+        private async Task RunChangeConsumptionLoopAsync()
+        {
+            try
+            {
+                CancellationToken token = this._cancellationTokenSourceCheckForChanges.Token;
+
+                using (var connection = new SqlConnection(this._connectionString))
+                {
+                    await connection.OpenAsync(token);
+
+                    while (!token.IsCancellationRequested)
+                    {
+                        if (this._state == State.CheckingForChanges)
+                        {
+                            // What should we do if this call gets stuck?
```

Comment:
What does "stuck" here mean? If it's taking a long time to execute then it will eventually time out and throw an exception so it should be handled just like any other exception that occurs when running these queries.

Reply:
I will remove the comment as there is handling added for this. Currently the `MaxRenewalCount` was 5 and taking into account 15 seconds or lease renewal interval, the lease will be release after 75 seconds. I am updating the `MaxRenewalCount` to 10 as 75 seconds seem to short. @Charles-Gagnon, in your comment, do you mean we should have a different strategy for dealing with functions that are stuck?

Reply:
I don't know about a different strategy - but we should make sure we handle that scenario deliberately. Right now will it actually get "stuck" or does `ITriggeredFunctionExecutor.TryExecuteAsync` have some sort of timeout that will throw an exception?

In general that would usually imply user error so I'm less concerned about handling that immediately - but this would be another good item to add as a follow-up to investigate.